docs/guillotine.adoc

```diff
@@ -0,0 +1,1349 @@
+
+.site/content-types/movie/movie.xml:
+[source,xml,options="nowrap"]
+----
+<content-type xmlns="urn:enonic:xp:model:1.0">
+  <display-name>Movie</display-name>
+  <description>Moving images often reflecting culture</description>
+  <super-type>base:structured</super-type>
+
+  <form>
+    <input name="image" type="ImageSelector">
+        <label>Movie poster</label>
+        <occurrences minimum="1" maximum="1"/>
+    </input>
+
+    <input name="description" type="HtmlArea">
+        <label>Description</label>
+        <config>
+            <exclude>*</exclude>
+            <include>Bold Italic Underline</include>
+        </config>
+    </input>
+
+    <input name="year" type="Long">
+        <label>Release year</label>
+        <occurrences minimum="1" maximum="1"/>
+    </input>
+
+    <input name="actor" type="TextLine">
+        <label>Actor</label>
+        <occurrences minimum="0" maximum="0"/>
+    </input>
+  </form>
+</content-type>
+----
+
+{zwsp} +
+
+=== React components
+
+Next, we'll set up a few react components for visualizing each movie item.
+
+The *entry*, _Movie.jsx_, will take care of rendering a preview of each movie content item in content studio later:
+
+.react4xp/myEntries/Movie.jsx:
+[source,javascript,options="nowrap"]
+----
+import React from 'react'
+
+import Movie from '../shared/Movie';
+
+export default (props) => <Movie {...props} />;
+----
+
+This is a pure entry wrapper that just imports the next react component from _react4xp/shared_.
+
+Why import code from _shared_ instead of keeping it all in the entry? Two reasons:
+
+- It's a good rule of thumb to keep entries slim, for better optimization.
+- Also, in addition to a Content Studio preview for single movies, we're going to use the *imported components* in the actual movie list too, for each single movie in the list. This way, the preview in Content Studio will always directly reflect what's displayed on the final page, because it's the same code that's used everywhere:
+
+.react4xp/shared/Movie.jsx:
+[source,javascript,options="nowrap"]
+----
+import React from 'react'
+
+import './Movie.scss';
+
+const Cast = ({actors}) => (
+    <ul className="cast">
+        { actors.map( actor => <li key={actor} className="actor">{actor}</li> ) }
+    </ul>
+);
+
+
+const Info = ({heading, children}) => (
+    <div className="info">
+        {heading ? <h3>{heading}</h3> : null}
+        {children}
+    </div>
+);
+
+
+const InfoContainer = ({title, year, description, actors}) => (
+    <div className="infoContainer">
+        <h2 className="title">{title}</h2>
+
+        <Info heading="Released">
+            <p className="year">{year}</p>
+        </Info>
+
+        <Info heading="Description">
+            <div className="description"
+                 dangerouslySetInnerHTML={{ __html: description }}>
```

Comment:
Yes, it's the standard way - equivalent to thymeleaf's `data-th-utext`. Strictly speaking it's not a necessary part of this lesson, I just thought it would be interesting to show that this context handles that too. Hmm, scary, I guess it's supposed to be... :D Maybe worth a comment at least?

Reply:
No need for more comments everything is so defailed allready. We probably want people to learn how to use html area.

Reply:
@sigdestad , same issue here.

Reply:
❓

Reply:
-> https://github.com/enonic/doc-react4xp/issues/7 
And remove htmlarea for now.